batching/batching.go

```diff
@@ -99,7 +100,7 @@ func (b Batches) Iter(ctx context.Context) <-chan data.Entry {
 	go func() {
```

Comment:
Note: If we want to pump this version to require 1.23, we could use the new range over stuff to avoid having an Iter() function at all, remove the need for the channel.